controller/engine_controller.go

```diff
@@ -620,7 +621,9 @@ func (ec *EngineController) stopMonitoring(e *longhorn.Engine) {
 		logrus.Warnf("engine %v: stop monitoring called when there is no monitoring", e.Name)
 		return
 	}
-	stopCh <- struct{}{}
+	if len(stopCh) == 0 {
```

Comment:
This might work in this situation (since the buffer size is 2 instead of 1), but this looks like a race condition at first glance since stopCh can receive another signal after `len` check but before the next line.

Reply:
This check is used to prevent this function from keeping signaling the channel. It's not for solving the race condition.